src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -296,6 +296,191 @@ EnsureModificationsCanRun(void)
 }
 
 
+/*
+ * IsNonDistributedTable returns whether the given relation ID identifies a
+ * non-distributed table such as reference table.
+ */
+bool
+IsNonDistributedTable(Oid relationId)
+{
+	if (!IsCitusTable(relationId))
```

Comment:
`IsCitusTable` and `IsNonDistributedTableCacheEntry ` may rely on different `CitusTableCacheEntry `. What happens if `GetCitusTableCacheEntry ` invalidates the cache entry in between? Would that be OK for all the functions/code-paths?